opencog/atoms/execution/EvaluationLink.cc

```diff
@@ -516,7 +516,12 @@ TruthValuePtr EvaluationLink::do_eval_scratch(AtomSpace* as,
 				continue;
 			}
 			if (not nameserver().isA(h->get_type(), FUNCTION_LINK))
-				throw NotEvaluatableException();
+			{
+				if (silent)
```

Comment:
I would suggest moving all code block below into separate function and call it here and below to remove code duplication:
```c++
void throwSyntaxException(bool silent, const char* message)
{
  if (silent)
    throw NotEvaluatableException();
  throw SyntaxException(TRACE_INFO, message);
}
```

Reply:
done